### PR TITLE
IRK

### PR DESCRIPTION
**Pull request by EveCharbie**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document [docs/contribution.md]?
* [x] Have you checked to ensure there aren't other open [Pull Requests] for the same update/change?
* [x] Have you opened/linked the issue related to your pull request?
* [x] Have you used the tag [WIP] for on-going changes, and removed it when the pull request was ready?
* [x] When ready to merge, have you sent a comment pinging @pariterre in it?

### New Feature Submissions:

1. [x] Does your submission pass the tests (if not please explain why this is intended)?
2. [x] Have you linted your code locally prior to submission (using the command: `black . -l120 --exclude "external/*"`)?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new examples for your core changes, as applicable?
* [ ] Have you written new tests for your core changes, as applicable?

### CODE REVIEW

**Comment by EveCharbie** [comment]
@pariterre I think the tests for IRK should cover more functionnalities than it is the case now. I would like to know if you have preferences otherwise I would just add "@pytest.mark.parametrize("ode_solver", [OdeSolver.RK, OdeSolver.IRK])" to almost every test.

File: bioptim/misc/enums.py [hunk 1]
```diff
@@ -27,6 +27,7 @@ class OdeSolver(Enum):
     RK = 1
     CVODES = 2
     NO_SOLVER = 3
+    IRK = 4
```
Comment (pariterre) [inline bioptim/misc/enums.py:30]: I'd put IRK in second place here

File: bioptim/misc/optimal_control_program.py [hunk 2]
```diff
@@ -20,7 +20,7 @@
 from ..dynamics.dynamics_type import DynamicsTypeList, DynamicsTypeOption
 from ..gui.plot import CustomPlot
 from ..interfaces.biorbd_interface import BiorbdInterface
-from ..interfaces.integrator import RK4
+from ..interfaces.integrator import RK4, IRK
```
Comment (pariterre) [inline bioptim/misc/optimal_control_program.py:23]: Is this an IRK4? Or the 4 would not make sense?

**Review by pariterre — CHANGES_REQUESTED** [review]
gg

**Comment by pariterre** [comment]
@EveCharbie 
I think adding this to almost all function does make sense. I don't think we need to test it in other ways than that, unless you expect IRK to behave in a very weird manner sometimes

**Comment by pariterre** [comment]
@EveCharbie 
The failing in black is probably due to the fact that your version of black is out-of-date. Make sure you update it (conda update black -cconda-forge) to the 20.8b1 at least

File: bioptim/misc/optimal_control_program.py (hunk 2 above)
Comment (EveCharbie) [reply bioptim/misc/optimal_control_program.py:23]: No it is an IRK of order 'degree' (option for the user)

**Comment by pariterre** [comment]
@EveCharbie 
I updated your code to the last refactor I've done because I wasn't sure how hard it would be to merge

**Comment by EveCharbie** [comment]
@pariterre, I think it's ready for you to request changes :)

**Comment by pariterre** [comment]
Black did not pass, make sure you are blacking with the latest version (20.8b1)

**Review by pariterre — APPROVED** [review]
GG
I'll manage the remaining changes myself!

**Comment by pariterre** [comment]
LGTM :D

**Pull request merged**